server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -176,6 +176,15 @@ public boolean expandWildcardsClosed() {
         return expandWildcards.contains(WildcardStates.CLOSED);
     }
 
+    /**
+     * @return whether wildcard expression should get expanded
+     */
+    public boolean expandWildcardExpressions() {
+        // only check open/closed since if we do not expand to open or closed it doesn't make sense to
+        // expand to hidden
+        return expandWildcardsOpen() || expandWildcardsClosed();
+    }
+
```

Comment:
I've added this new options-test method, to make it easier to track all the code places that rely on wildcards being expanded or not.